pkg/cluster/kubernetes/sync_test.go

```diff
@@ -464,6 +485,18 @@ metadata:
 		test(t, kube, "", ns1+defs1+defs2+ns3+defs3, false)
 	})
 
+	t.Run("sync adds and GCs skips resources", func(t *testing.T) {
```

Comment:
This test does verify what it claims to; but I think it should also verify that resources are updated (i.e., not ignored) when they have a gc_disabled annotation.